### DISCUSSION TITLE
Hopper Upgrade

### DISCUSSION THREAD

**Discussion by digitalbarrito** [P3pp3rF1y/SophisticatedBackpacks#142] [Ideas]
Upgrade that gives the backpack hopper functionality and an Advanced Version with Filtering/Better Filtering. This would fit in nicely with the filtering upgrade. The idea behind this is mainly for Auto-Sorting Storage systems, with this it would be feasible to cut out the many sorting hoppers and allow for much better Auto-Sorting Storage builds imo.

**Comment by P3pp3rF1y** [comment]
I am going to look into adding filter by backpack contents in the filter upgrade which should help quite a bit. 

as for hopper upgrade I don't see a good reason to add it - pickup upgrade allows to pickup item entities above backpack and for piping in / out it makes much more sense to me to use hopper in vanilla like experience and in modded there are much better ways of piping in/out where this upgrade wouldn't make any sense
